snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/dsv/SnomedSimpleTypeRefSetDSVExporter.java

```diff
@@ -544,408 +530,146 @@ private void createHeaderList() throws SQLException {
+					Integer count = (int) concept.getRelationships()
+									.getItems().stream()
+									.flatMap(relationship -> relationship.getMembers().getItems().stream())
+									.filter(cdMember -> cdMember.getProperties().get(Fields.ATTRIBUTE_NAME).toString().contains(item.getDisplayName()))
+									.count();
+					if (result.containsKey(item.getDisplayName())) {
+						result.compute(item.getDisplayName(), (key, oldCountValue) -> Math.max(oldCountValue, count));
+					} else {
+						result.put(item.getDisplayName(), count);
 					}
-					
-					return null;
+//					result.merge(item.getDisplayName(), count,  (key, oldCountValue) -> Math.max(oldCountValue, count));
+					break;
 				}
-			});
-			
-		}
-
-		preferredTermStatement.setString(i++, String.valueOf(languageConfiguration));
-
-		preferredTermStatement.setInt(i++, branchId);
-		preferredTermStatement.setInt(i++, branchId);
-		preferredTermStatement.setLong(i++, branchBase);
-		preferredTermStatement.setLong(i++, branchBase);
-
-		resultSet = preferredTermStatement.executeQuery();
-		resultSet.absolute(1);
-		String result;
-		if (resultSet.isFirst()) {
-			result = resultSet.getString(1);
-			if (descriptionIdExpected) {
-				result = resultSet.getString(3) + delimiter + result;
+				default: break;
 			}
-		} else {
-			result = "";
-			if (descriptionIdExpected) {
-				result += delimiter;
 			}
 		}
 
 		return result;
 	}
 
+	private ISnomedConceptNameProvider getNameProvider() {
+		return ApplicationContext.getServiceForClass(ISnomedConceptNameProvider.class);
+	}
+
+
 	/**
-	 * Returns the effective time of the exported concept.
+	 * This method implements a maximum search. It founds the maximum presence of a given description, relationship or concrete datatype.
+	 * @param occurenceByTypeId 
+	 * @param members 
 	 * 
-	 * @param conceptCDOId
+	 * @param queryType
+	 *            - type of the query can be description, relationship and data type.
+	 * @param typeId
+	 *            - type id of the description or relationship or data type.
 	 * @return
-	 * @throws SQLException
 	 */
-
-	private String getEffectiveTime(long conceptCDOId) throws SQLException {
-		PreparedStatement effectiveTimeStatement = connection.prepareStatement(SnomedRefSetExporterQueries.SQL_SIMPLE_TYPE_REFSET_DSV_EXPORT_EFFECTIVE_TIME_QUERY);
-		effectiveTimeStatement.setLong(1, conceptCDOId);
-		effectiveTimeStatement.setInt(2, branchId);
-		effectiveTimeStatement.setInt(3, branchId);
-		effectiveTimeStatement.setLong(4, branchBase);
-		effectiveTimeStatement.setLong(5, branchBase);
-		ResultSet resultSet = effectiveTimeStatement.executeQuery();
-		if (resultSet.absolute(1)) {
-			Date effectiveTime = resultSet.getDate(1);
-			if (null == effectiveTime) {
-				return EffectiveTimes.UNSET_EFFECTIVE_TIME_LABEL;
-			}
-			return effectiveTime.toString();
-		} else {
-			return "";
-		}
-
+	private int getMaxOccurence(Map<String, Integer> occurenceByTypeId, String typeId) {
```

Comment:
I think over the whole set of members a Refset can have, 
to find the maximal amount of description/relationship types of all members:
the Map with compute/merge methods and Max checking of present values, would result in a more self explanatory source code